init.el

```diff
@@ -25,16 +25,16 @@
 (load (concat spacemacs-core-directory "core-dumper.el")
       nil (not init-file-debug))
 
-;; Clean compiled files if they become stale or Emacs version has changed.
+;; Remove compiled core files if they become stale or Emacs version has changed.
 (load (concat spacemacs-core-directory "core-compilation.el")
       nil (not init-file-debug))
 (load spacemacs--last-emacs-version-file t (not init-file-debug))
-(let ((default-directory spacemacs-start-directory))
-  (when (or (not (string= spacemacs--last-emacs-version emacs-version))
-            (spacemacs//contains-newer-than-byte-compiled-p
-             spacemacs--compiled-files))
-    (spacemacs//remove-byte-compiled-files spacemacs--compiled-files)))
-(when (not (string= spacemacs--last-emacs-version emacs-version))
+(when (or (not (string= spacemacs--last-emacs-version emacs-version))
+          (spacemacs//dir-contains-stale-byte-compiled-files-p
+           spacemacs-core-directory))
+  (spacemacs//remove-byte-compiled-files-in-dir spacemacs-core-directory))
```

Comment:
I remove all the .elc files to prevent any kind of macro dependency shenanigans by heaving clean compilation afterwards.